.github/workflows/publish-gem.yml

```diff
@@ -0,0 +1,23 @@
+name: Publish Gem
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7.8'
+      - name: Build gem
+        run: |
+          gem build github-pages.gemspec
+      - name: Publish
```

Comment:
this should only run when merge into master branch I think?

Reply:
Do we want to instead run the job when we merge into `master` branch?
```yaml
on:
  push:
    branches:
      - main
```

Reply:
```
on:
  release:
    types: [released]
```